spec/features/users_show_spec.rb

```diff
@@ -0,0 +1,33 @@
+require 'rails_helper'
+
+RSpec.describe 'User Show Page', type: :feature do
+  before(:each) do
+    @user1 = User.create(name: 'Tom', photo: 'https://unsplash.com/photos', bio: 'Teacher from Mexico.',
+                         post_counter: 0)
+    @first_post = Post.create(author: @user1, title: 'post1', text: 'This is my first post', like_counter: 0,
+                              comment_counter: 0)
+    @second_post = Post.create(author: @user1, title: 'post2', text: 'This is my second post', like_counter: 0,
+                               comment_counter: 0)
+    @third_post = Post.create(author: @user1, title: 'post3', text: 'This is my third post', like_counter: 0,
+                              comment_counter: 0)
+    @fourth_post = Post.create(author: @user1, title: 'post4', text: 'This is my fourth post', like_counter: 0,
+                               comment_counter: 0)
+    visit "/users/#{@user1.id}"
+  end
+
+  it "should displays the user's profile picture" do
+    expect(page).to have_css("img[src*='#{@user1.photo}']")
+  end
+
+  it 'should displays the username' do
+    expect(page).to have_content(@user1.name)
+  end
+
+  it 'should displays the number of posts ' do
+    expect(page).to have_content(@user1.post_counter)
+  end
+
+  it "should displays the user's bio" do
+    expect(page).to have_content(@user1.bio)
+  end
+end
```

Comment:
Well done so far on writing all of your test cases. However, in the user_show_spec page, you still haven't implemented some of the required tests. 🙏 

- [ ] Please write a test that shows the user's first 3 posts
> Note that this tested user should have more than 3 posts to ensure your test works fine when it shows only the first 3 on this user show page.

- [ ] Please write a test that shows the button 'see all posts` for that user
- [ ] For the 3 shown posts on the user show page, please write a test that redirects the user to a post's show page when one of these posts is clicked
- [ ] Please write a test that redirects to the user's post's index page when you click on the see all posts button.

_You can recheck for those requirements starting from this one right [here](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/projects/intergration_spec_and_n_1_project.md#:~:text=I%20can%20see%20the%20user%27s%20first%203%20posts.)._ 🙏